lib/connectionConfig/connectionConfig.ts

```diff
@@ -112,25 +113,25 @@ export namespace ConnectionConfig {
    */
   export function validate(config: ConnectionConfig, options?: ConnectionConfigOptions): void {
     if (!options) options = {};
-    if (!config || (config && typeof config !== "object")) {
-      throw new Error("'config' is a required parameter and must be of type: 'object'.");
-    }
-    if (!config.endpoint || (config.endpoint && typeof config.endpoint !== "string")) {
-      throw new Error("'endpoint' is a required property of ConnectionConfig.");
-    }
-    if (config.entityPath && typeof config.entityPath !== "string") {
-      throw new Error("'entityPath' must be of type 'string'.");
-    }
-    if (options.isEntityPathRequired && !config.entityPath) {
-      throw new Error("'entityPath' is a required property of ConnectionConfig.");
-    }
-    if (!config.sharedAccessKeyName ||
-      (config.sharedAccessKeyName && typeof config.sharedAccessKeyName !== "string")) {
-      throw new Error("'sharedAccessKeyName' is a required property of ConnectionConfig.");
+
+    throwTypeErrorIfParameterMissing("config", config);
+    throwTypeErrorIfParameterTypeMismatch("config", config, "object");
+
+    throwTypeErrorIfParameterMissing("endpoint", config.endpoint);
+    config.endpoint = String(config.endpoint);
+
+    if (config.entityPath) {
+      config.entityPath = String(config.entityPath);
     }
-    if (!config.sharedAccessKey ||
-      (config.sharedAccessKey && typeof config.sharedAccessKey !== "string")) {
-      throw new Error("'sharedAccessKey' is a required property of ConnectionConfig.");
+
+    if (options.isEntityPathRequired) {
+      throwTypeErrorIfParameterMissing("entityPath", config.entityPath);
     }
+
+    throwTypeErrorIfParameterMissing("sharedAccessKeyName", config.sharedAccessKeyName);
+    config.sharedAccessKeyName = String(config.sharedAccessKeyName);
+
+    throwTypeErrorIfParameterMissing("sharedAccessKey", config.sharedAccessKey);
+    config.sharedAccessKey = String(config.sharedAccessKey);
```

Comment:
Errors like `Missing parameter "abc"` from a `validate` method doesnt give good enough information to the end users.
In the changes we did in service bus, we had such errors only in the user facing methods. In that case the word "parameter" in error message makes sense. There was a problem in the parameter they provided to a method.

Since most of the functions in this library will not be user facing and instead are internal, we need to rethink the error message.

- `Missing configuration`
- `Missing "endpoint" in configuration`
- `Missing "entityPath" in configuration`
- `Missing "sharedAccessKeyName" in configuration`
- `Missing "sharedAccessKey" in configuration`